contrib/rust/src/python/pants/contrib/rust/utils/custom_build_invocation/BUILD

```diff
@@ -0,0 +1,51 @@
+target(
+  dependencies=[
+    ':args_rules',
+    ':env_rules',
+    ':links_rule',
+    ':outputs_rule',
+    ':program_rule',
+  ]
+)
+
```

Comment:
There is a huge amount to absorb in this PR... would it be possible to break out some of the more advanced features of the cargo support into a second PR? I'm commenting here just because I'm not sure I recognize "custom build invocations" (...are they `build.rs` scripts?)